rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java

```diff
@@ -128,9 +140,20 @@ public BuildRecordRest(BuildRecord buildRecord) {
         this.distributedMilestoneBuildRecordSetIds = nullableStreamOf(buildRecord.getBuildRecordSets())
                 .filter(buildRecordSet -> buildRecordSet.getDistributedInProductMilestone() != null)
                 .map(buildRecordSet -> buildRecordSet.getId()).collect(Collectors.toSet());
+
+        // To prevent lazy loading of collections, go through the transformation chain
```

Comment:
Can we hide complex ORM code in DatastoreAdapter? *Rest DTOs should be as simple as possible and deal mostly with (de)serialization.